crates/sdk/src/tx.rs

```diff
@@ -2003,21 +2003,45 @@ pub async fn build_vote_proposal(
         }
     }
 
-    let delegations = rpc::get_delegators_delegation_at(
-        context.client(),
-        voter,
-        proposal.voting_start_epoch,
-    )
-    .await?
-    .keys()
-    .cloned()
-    .collect::<Vec<Address>>();
-
-    if delegations.is_empty() {
-        return Err(Error::Other(
-            "Voter address must have delegations".to_string(),
-        ));
-    }
+    let delegations = if is_validator {
+        let current_epoch = query_epoch(context.client()).await?;
```

Comment:
dont need to query for the epoch, can you the one from received in the parameters